test/onnx/test_op_consistancy.py

```diff
@@ -0,0 +1,461 @@
+    dont_care(
+        "ceil", dtypes=BOOL_TYPES + INT_TYPES + QINT_TYPES + COMPLEX_TYPES,
+        reason=reason_onnx_does_not_support("Ceil")
+    ),
+    skip("ceil", dtypes=[torch.float64], reason=reason_onnx_runtime_does_not_support("Ceil", ["f64"])),
+    dont_care("sqrt", dtypes=BOOL_TYPES + QINT_TYPES + COMPLEX_TYPES, reason=reason_onnx_does_not_support("Sqrt")),
+    xfail("t", dtypes=COMPLEX_TYPES, reason=reason_jit_tracer_error("complex types")),
+)
+# fmt: on
+
+# Expected opset specific fails for ops that do not support specific opsets
+
+EXPECTED_OPSET_FAILS: Tuple[XfailOpset, ...] = (
+    # TODO: sqrt for torch.bfloat16 is just an example. Replace it with more meaningful
+    # skips when there are.
+    XfailOpset(
+        "sqrt",
+        dtypes=[torch.bfloat16],
+        opsets=[opsets_before(13)],
+        reason="Sqrt not defined for bf16 before opset 13",
+    ),
+)
+
+# END OF SECTION TO MODIFY #####################################################
+
+
+OPS_DB = copy.deepcopy(common_methods_invocations.op_db)
+
+
+class SingleOpModel(torch.nn.Module):
+    """Test model to wrap around a single op for export."""
+
+    def __init__(self, op, kwargs):
+        super().__init__()
+        self.operator = op
+        self.kwargs = kwargs
+
+    def forward(self, *args):
+        return self.operator(*args, **self.kwargs)
+
+
+class TestConsistency(common_utils.TestCase):
+    """Test consistency of exported ONNX models.
+
+    This is a parameterized test suite.
+    """
+
+    @common_device_type.ops(OPS_DB, allowed_dtypes=SUPPORTED_DTYPES)
+    @skip_ops(
+        OPS_DB,
+        "TestConsistency",
+        "test_output_match",
+        skip_or_xfails=EXPECTED_SKIPS_OR_FAILS,
+    )
+    def test_output_match(self, device: str, dtype: torch.dtype, op):
+        assert device == "cpu"
+
+        if op.name not in ALLOWLIST_OP:
+            self.skipTest(f"'{op.name}' is not in the allow list for test on ONNX")
+
+        expected_opset_fails_name_mapping = {
+            fail.op_name: fail for fail in EXPECTED_OPSET_FAILS
+        }
+        samples = op.sample_inputs(
+            device,
+            dtype,
+            requires_grad=False,
+        )
+
+        for i, (opset, cpu_sample) in enumerate(
+            itertools.product(TESTED_OPSETS, samples)
+        ):
+            # Provide the repr to subtest because tensors are not serializable in parallel test runs
+            with self.subTest(
+                opset=opset,
+                sample_num=i,
+                input=repr(cpu_sample.input),
+                args=repr(cpu_sample.args),
+                kwargs=repr(cpu_sample.kwargs),
+            ):
+                model = SingleOpModel(op, cpu_sample.kwargs)
+                model.eval()
+
+                context_manager = contextlib.nullcontext()
+                # Skip opset specific fails
+                if op.name in expected_opset_fails_name_mapping:
```

Comment:
Can we simplify the logic here and replace these two context managers with similar decorator as `skip_ops`, or merge with `skip_ops`? It looks like it's only because opset is not accessible at top level test case. Is there special consideration to not do that?

Reply:
It's because opset is not accessible, and also because skip_ops uses the DecorateInfo and modifies the DB directly. The parameterization logic in `common_device_type.ops` then uses the DecorateInfo to decide what to skip, which doesn't have access to opset information.

Reply:
I thought of parameterizing the test class, but then I am not sure how to interact with `common_device_type.instantiate_device_type_tests` with that

Reply:
Link: https://github.com/pytorch/pytorch/pull/87536